### DISCUSSION TITLE
SB 5.1 OCI Provision/DataPipeline/InfluxDB failing public key/signature error

### DISCUSSION THREAD

**Discussion by subhashchandrab** [Sunbird-Ed/Community#411] [Installation]
Following the documentation https://ed.sunbird.org/use/installing-ed-platform/release-5.1.0/build. The Jenkins job Provision/DataPipeline/InfluxDB is failing with below error. This issue seems to be related to https://www.influxdata.com/blog/linux-package-signing-key-rotation/. Can you please check and confirm

TASK [influxdb : Add InfluxDB repository] **************************************
fatal: [10.x.x.x]: FAILED! => {
    "changed": false,
    "rc": 1
}

MSG:

MODULE FAILURE
See stdout/stderr for the exact error

MODULE_STDOUT:

Traceback (most recent call last):

  ```
File "/home/ubuntu/.ansible/tmp/ansible-tmp-1676568086.3408206-292155-207505551859988/AnsiballZ_apt_repository.py", line 114, in <module>

    _ansiballz_main()

  File "/home/ubuntu/.ansible/tmp/ansible-tmp-1676568086.3408206-292155-207505551859988/AnsiballZ_apt_repository.py", line 106, in _ansiballz_main

    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)

  File "/home/ubuntu/.ansible/tmp/ansible-tmp-1676568086.3408206-292155-207505551859988/AnsiballZ_apt_repository.py", line 49, in invoke_module

    imp.load_module('__main__', mod, module, MOD_DESC)

  File "/tmp/ansible_apt_repository_payload_9TpB2S/__main__.py", line 554, in <module>

  File "/tmp/ansible_apt_repository_payload_9TpB2S/__main__.py", line 546, in main

  File "/usr/lib/python2.7/dist-packages/apt/cache.py", line 591, in update

    raise FetchFailedException(e)

apt.cache.FetchFailedException: W:GPG error: https://repos.influxdata.com/ubuntu trusty InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY D8FF8E1F7DF8B07E, E:The repository 'https://repos.influxdata.com/ubuntu trusty InRelease' is not signed.
```

MODULE_STDERR:

Shared connection to 10.x.x.x closed.

**Comment by subhashchandrab** [comment]
Ths target VM in this case is Ubuntu 20.02. Was able to resolve it by changing the key in  https://github.com/Sunbird-Obsrv/sunbird-data-pipeline/blob/release-5.1.0/ansible/roles/influxdb/tasks/main.yml as suggested in https://www.influxdata.com/blog/linux-package-signing-key-rotation/.  

    - name: "Import InfluxDB GPG signing key"
      apt_key: url=https://repos.influxdata.com/influxdata-archive_compat.key

**Comment by subhashchandrab** [comment]
After installation went fine, it started giving the below error

fatal: [10.x.x.x]: FAILED! => {
    "changed": false,
    "cmd": "/usr/bin/influx -execute 'CREATE DATABASE monitoring_events'",
    "rc": 2
}

MSG:

[Errno 2] No such file or directory

Looks like the influxdb pacakge installation doesn't automatically install the influx cli. It's resolved by adding the following task in  https://github.com/Sunbird-Obsrv/sunbird-data-pipeline/blob/release-5.1.0/ansible/roles/influxdb/tasks/main.yml

- name: Install InfluxDB client packages
  apt: name=influxdb-client state=present
  notify: restart influxdb service

**Reply by vinukumar-vs** [reply]
@anandp504 Can you please help here.

**Reply by chitranshu-keshav** [reply]
@anandp504 @manjudr can you please get back on this
